File: leetcode/238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.java
// Last updated: 29/04/2026, 23:11:18
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];
        int left = 1;
        for(int i=0; i<nums.length; i++) {
            if(i>0) left = left * nums[i-1];
            res[i] = left;
        }
        int right = 1;
        for(int i=nums.length-1; i>=0; i--) {
            if(i<nums.length-1) right = right*nums[i+1];
            res[i]*=right;
        }
        return res;
    }
}

File: leetcode/238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 29/04/2026, 23:11:12
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums) 
        pre = 1
        for i in range(len(nums)):
            temp = nums[i]
            res[i] = pre
            pre = temp * pre
        
        post = 1
        for i in range(len(nums)-1,-1,-1):
            temp = nums[i]
            res[i] = post * res[i]
            post = temp * post
        return res